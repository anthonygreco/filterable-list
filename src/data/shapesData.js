export default [
  {
    name: 'a',
    type: 'square',
    info: {
      length: 1.0
    }
  },
  {
    name: 'b',
    type: 'circle',
    info: {
      radius: 1.0
    }
  },
  {
    name: 'c',
    type: 'triangle',
    info: {
      angles: [
        20.0, 40.0, 20.0
      ]
    }
  },
  {
    name: 'd',
    type: 'square',
    info: {
      length: 2.0
    }
  },
  {
    name: 'e',
    type: 'circle',
    info: {
      radius: 2.0
    }
  },
  {
    name: 'f',
    type: 'triangle',
    info: {
      angles: [
        10.0, 100.0, 20.0
      ]
    }
  },
  {
    name: 'g',
    type: 'square',
    info: {
      length: 3.0
    }
  },
  {
    name: 'h',
    type: 'circle',
    info: {
      radius: 3.0
    }
  },
  {
    name: 'i',
    type: 'triangle',
    info: {
      angles: [
        60.0, 30.0, 40.0
      ]
    }
  },
  {
    name: 'j',
    type: 'square',
    info: {
      length: 4.0
    }
  },
  {
    name: 'k',
    type: 'circle',
    info: {
      radius: 4.0
    }
  },
  {
    name: 'l',
    type: 'triangle',
    info: {
      angles: [
        20.0, 50.0, 60.0
      ]
    }
  },
  {
    name: 'm',
    type: 'square',
    info: {
      length: 5.0
    }
  },
  {
    name: 'n',
    type: 'circle',
    info: {
      radius: 5.0
    }
  },
  {
    name: 'o',
    type: 'triangle',
    info: {
      angles: [
        40.0, 20.0, 40.0
      ]
    }
  },
  {
    name: 'p',
    type: 'square',
    info: {
      length: 6.0
    }
  },
  {
    name: 'q',
    type: 'circle',
    info: {
      radius: 6.0
    }
  },
  {
    name: 'r',
    type: 'triangle',
    info: {
      angles: [
        30.0, 30.0, 60.0
      ]
    }
  },
  {
    name: 's',
    type: 'square',
    info: {
      length: 7.0
    }
  },
  {
    name: 't',
    type: 'circle',
    info: {
      radius: 7.0
    }
  },
  {
    name: 'u',
    type: 'triangle',
    info: {
      angles: [
        10.0, 40.0, 30.0
      ]
    }
  },
  {
    name: 'v',
    type: 'square',
    info: {
      length: 8.0
    }
  },
  {
    name: 'w',
    type: 'circle',
    info: {
      radius: 8.0
    }
  },
  {
    name: 'x',
    type: 'triangle',
    info: {
      angles: [
        20.0, 30.0, 90.0
      ]
    }
  },
  {
    name: 'y',
    type: 'square',
    info: {
      length: 9.0
    }
  },
  {
    name: 'z',
    type: 'circle',
    info: {
      radius: 9.0
    }
  }
]
