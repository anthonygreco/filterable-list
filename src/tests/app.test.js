import React from 'react'
import ReactDOM from 'react-dom'
import { render } from 'react-testing-library'

import App from '../components/app'
import { Container } from 'semantic-ui-react'
import FilterableList from '../components/filterableList'
import shapesData from '../data/shapesData'

const filterProps = ['name', 'type']

test('<App /> renders without crashing', async () => {
  const div = document.createElement('div')

  ReactDOM.render(<App />, div)
})

test('<App /> should produce a proper snapshot', async () => {
  const { container } = render(
    <App>
      <Container fluid>
        <FilterableList entries={shapesData} filterProps={filterProps} />
      </Container>
    </App>
  )

  expect(container).toMatchSnapshot()
})
