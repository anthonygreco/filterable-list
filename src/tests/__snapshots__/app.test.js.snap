// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should produce a proper snapshot 1`] = `
<div>
  <div
    class="app"
  >
    <div
      class="ui fluid container"
    >
      <div>
        <div
          class="ui fluid icon left corner labeled input"
        >
          <div
            class="ui label label left corner"
          >
            <i
              aria-hidden="true"
              class="asterisk icon"
            />
          </div>
          <input
            type="text"
            placeholder="Filter..."
          />
          <i
            aria-hidden="true"
            class="search icon"
          />
        </div>
        <table
          class="ui selectable striped table"
        >
          <thead
            class=""
          >
            <tr
              class=""
            >
              <th
                class=""
              >
                Name
              </th>
              <th
                class=""
              >
                Type
              </th>
              <th
                class=""
              >
                Info
              </th>
            </tr>
          </thead>
          <tbody
            class=""
          >
            <tr
              class=""
            >
              <td
                class=""
              >
                a
              </td>
              <td
                class=""
              >
                square
              </td>
              <td
                class=""
              >
                {length:1}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                b
              </td>
              <td
                class=""
              >
                circle
              </td>
              <td
                class=""
              >
                {radius:1}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                c
              </td>
              <td
                class=""
              >
                triangle
              </td>
              <td
                class=""
              >
                {angles:[20,40,20]}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                d
              </td>
              <td
                class=""
              >
                square
              </td>
              <td
                class=""
              >
                {length:2}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                e
              </td>
              <td
                class=""
              >
                circle
              </td>
              <td
                class=""
              >
                {radius:2}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                f
              </td>
              <td
                class=""
              >
                triangle
              </td>
              <td
                class=""
              >
                {angles:[10,100,20]}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                g
              </td>
              <td
                class=""
              >
                square
              </td>
              <td
                class=""
              >
                {length:3}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                h
              </td>
              <td
                class=""
              >
                circle
              </td>
              <td
                class=""
              >
                {radius:3}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                i
              </td>
              <td
                class=""
              >
                triangle
              </td>
              <td
                class=""
              >
                {angles:[60,30,40]}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                j
              </td>
              <td
                class=""
              >
                square
              </td>
              <td
                class=""
              >
                {length:4}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                k
              </td>
              <td
                class=""
              >
                circle
              </td>
              <td
                class=""
              >
                {radius:4}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                l
              </td>
              <td
                class=""
              >
                triangle
              </td>
              <td
                class=""
              >
                {angles:[20,50,60]}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                m
              </td>
              <td
                class=""
              >
                square
              </td>
              <td
                class=""
              >
                {length:5}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                n
              </td>
              <td
                class=""
              >
                circle
              </td>
              <td
                class=""
              >
                {radius:5}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                o
              </td>
              <td
                class=""
              >
                triangle
              </td>
              <td
                class=""
              >
                {angles:[40,20,40]}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                p
              </td>
              <td
                class=""
              >
                square
              </td>
              <td
                class=""
              >
                {length:6}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                q
              </td>
              <td
                class=""
              >
                circle
              </td>
              <td
                class=""
              >
                {radius:6}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                r
              </td>
              <td
                class=""
              >
                triangle
              </td>
              <td
                class=""
              >
                {angles:[30,30,60]}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                s
              </td>
              <td
                class=""
              >
                square
              </td>
              <td
                class=""
              >
                {length:7}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                t
              </td>
              <td
                class=""
              >
                circle
              </td>
              <td
                class=""
              >
                {radius:7}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                u
              </td>
              <td
                class=""
              >
                triangle
              </td>
              <td
                class=""
              >
                {angles:[10,40,30]}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                v
              </td>
              <td
                class=""
              >
                square
              </td>
              <td
                class=""
              >
                {length:8}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                w
              </td>
              <td
                class=""
              >
                circle
              </td>
              <td
                class=""
              >
                {radius:8}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                x
              </td>
              <td
                class=""
              >
                triangle
              </td>
              <td
                class=""
              >
                {angles:[20,30,90]}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                y
              </td>
              <td
                class=""
              >
                square
              </td>
              <td
                class=""
              >
                {length:9}
              </td>
            </tr>
            <tr
              class=""
            >
              <td
                class=""
              >
                z
              </td>
              <td
                class=""
              >
                circle
              </td>
              <td
                class=""
              >
                {radius:9}
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
